### DISCUSSION TITLE
Action button

### DISCUSSION THREAD

**Discussion by alexmateescu** [dan-snelson/Nudge-Post-install#4] [Q&A]
Hi

So far i managed to deploy the software to my users and i also used the post-install script  to create the json and the launch agent. 

My question is how could i, instead of opening system preferences for the update to launch the "softwareupdate -i -a" command?

Cheers

**Comment by dan-snelson** [comment]
@alexmateescu The recommended approach is to have a Jamf Pro Policy which leverages the "File and Processes" payload to execute a command. See:
- https://docs.jamf.com/10.35.0/jamf-pro/documentation/Policy_Payload_Reference.html?hl=execute%2Ccommands
- https://github.com/dan-snelson/Nudge-Post-install/blob/main/Nudge-Post-install.bash#L356

**Comment by alexmateescu** [comment]
I am not using jamf. 

We are using another MDM so can’t apply that. 

Is there any other way to use that button ?

**Reply by dan-snelson** [reply]
Probably best to ask on the MacAdmins [#nudge](https://slack.com/app_redirect?channel=CDUU7DJQ2) channel about other ways to use [`actionButtonPath`](https://github.com/macadmins/nudge/wiki/userInterface#actionbuttonpath---type-string-default-value-).
